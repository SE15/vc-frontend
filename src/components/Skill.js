import { Spacer, Heading, Flex, Button, Box, HStack } from "@chakra-ui/react"
import { DeleteIcon} from '@chakra-ui/icons'
import {
    AlertDialog,
    AlertDialogBody,
    AlertDialogFooter,
    AlertDialogHeader,
    AlertDialogContent,
    AlertDialogOverlay 
  } from "@chakra-ui/react"

import {React,useState,useRef} from 'react' 


const Skill = (props) => {
    const [isOpen, setIsOpen] = useState(false)
    const onClose = () => setIsOpen(false)
    const cancelRef = useRef()




    return (
        <>
            <Flex>
                <Box w="20rem" p={4} bg="white.200">
                    <HStack  spacing="3rem">
                    <Heading size="md" color="black.600">Skill Name</Heading>
                
                <Spacer />
                
                   
                    <Button rightIcon={<DeleteIcon />} colorScheme="red" onClick={() => setIsOpen(true)}></Button>
                    <AlertDialog
        isOpen={isOpen}
        leastDestructiveRef={cancelRef}
        onClose={onClose}
      >
            <AlertDialogOverlay>
          <AlertDialogContent>
          <Box bg="white.200" w="80%" p={4} color="white">
            <AlertDialogHeader fontSize="lg" fontWeight="bold">
              Delete Skill
            </AlertDialogHeader>

            <AlertDialogBody>
              Are you sure? You can't undo this action afterwards.
            </AlertDialogBody>

            <AlertDialogFooter>
              <Button ref={cancelRef} onClick={onClose}>
                Cancel
              </Button>
              <Button colorScheme="red" onClick={onClose} ml={3}>
                Delete
              </Button>
            </AlertDialogFooter>
            </Box>
          </AlertDialogContent>
        </AlertDialogOverlay>
      </AlertDialog>
                    </HStack>
                </Box>
            </Flex>

        </>
    )
}

export default Skill;
