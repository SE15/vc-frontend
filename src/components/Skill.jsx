
//import { Link } from "react-router-dom"
import PropTypes from "prop-types"
import { Spacer, Heading, Flex, Button, Box, HStack ,CircularProgress} from "@chakra-ui/react"
import { DeleteIcon,AddIcon} from '@chakra-ui/icons'
import {
    Modal, ModalOverlay, ModalContent, Image
    , ModalCloseButton, Text, useDisclosure, ModalFooter,
     Textarea, Avatar, Stack
} from '@chakra-ui/react';
import {
    AlertDialog,Badge,
    AlertDialogBody,
    AlertDialogFooter,
    AlertDialogHeader,
    AlertDialogContent,
    AlertDialogOverlay 
  } from "@chakra-ui/react"

import {React,useState,useRef, useEffect} from 'react' 
import axios from 'axios';
import { kPrimaryBlack, kPrimaryBlackLight,kPrimaryGray, kSecondaryBlue, kSecondaryBlueDark, kSecondaryBlueLight } from './../utils/constants'
const Skill=props=>{
    
    const [isOpen, setIsOpen] = useState(false)
    const onClose = () => setIsOpen(false)
    const onOpen = () => setIsOpen(true)
    const cancelRef = useRef()
    const [name, setName] = useState(props.name);
    const [image, setimage]=useState(props.image)
    const [isLoading, setIsLoading] = useState(false);
    const [validations, setValidations]=useState(props.validations)
    const [skillname, setSkillname]=useState(props.skillname)
    
    const [error, setError] = useState('');

    const handleValidation= event=>{
        setIsLoading(true);
        try{
            const res=axios.put('api/users/skills/:id',{data:validations+1})
            setValidations(validations+1);
            setIsLoading(false);

        }catch{
            setValidations(validations);
            setIsLoading(false);
            setError("Couldn't validate skill")
        }
    }
    useEffect(()=>{
        handleValidation()
        console.log(validations)
    },[])

    
if(props.visit=="0"){
    
                
                    
    return (
        <>
        <Flex>
            <Box w="100%" p={4} bg={kPrimaryGray}>
                <HStack  spacing="3rem">
                        
                        
                    <Badge ml="1" fontSize="sm" colorScheme="blue" variant="outline" borderRadius="5rem">
                        {validations}
                    </Badge>
                          
                  
                          
                    <Heading size="sm" color={kSecondaryBlue}>{skillname}</Heading>
                    
                    <Spacer />
                
                   
                    <Button rightIcon={<DeleteIcon />} colorScheme="red"  onClick={() => setIsOpen(true)}></Button>
                  
                 
                
                        <AlertDialog
                                    isOpen={isOpen}
                                    leastDestructiveRef={cancelRef}
                                    onClose={onClose}
                                >
                            <AlertDialogOverlay>
                            <AlertDialogContent>
                            <Box bg={kPrimaryBlackLight} w="100%" p={4} color="white">

                            <AlertDialogHeader fontSize="lg" fontWeight="bold">
                            Delete Skill
                            </AlertDialogHeader>

                            <AlertDialogBody>
                            Are you sure? You can't undo this action afterwards.
                            </AlertDialogBody>

                            <AlertDialogFooter>
                            <Button colorScheme={kPrimaryBlack} ref={cancelRef} onClick={onClose}>
                                Cancel
                            </Button>
                            <Button colorScheme="red" onClick={onClose} ml={3}>
                            
                                Delete
                            </Button>
                            </AlertDialogFooter>

                            </Box>

                            </AlertDialogContent>
                            </AlertDialogOverlay>
                        </AlertDialog>
      
                    </HStack>
                    </Box>
                    </Flex>
                    </>
                   
        
    );
}else{
    return (
        <>
            <Flex>
                    <Box w="100%" p={4} bg={kPrimaryGray}>
                        <HStack  spacing="3rem">
                        <Badge ml="1" fontSize="0.8em" colorScheme="blue" variant="outline" borderRadius="5rem">
                            {validations}
                            </Badge>
                          
                        <Heading size="sm" color={kSecondaryBlue}>{skillname}</Heading>
                    
                    <Spacer />
                
                    {isLoading ? (
                     
                     <CircularProgress isIndeterminate color="green.300"/>
                   
                ) :(
                   <>
                    <Button rightIcon={<AddIcon />} colorScheme="blue" onClick={onOpen} size="sm" isDisabled={false}>
                    </Button>
                    <Modal isOpen={isOpen} onClose={onClose} color="black.700" closeOnOverlayClick="false">
                    <ModalOverlay />
                    <ModalContent>
                        <Box bg={kPrimaryBlackLight} w="100%" p={4} color="white">
                            <stack>
                                <Heading color="black.400" fontSize="lg">
                                    Skill Validation
                </Heading>
                                <br />
                                <Stack direction="row">
                                {props.cimage!=null?(
                                    <Image src={props.image}  borderRadius="full" boxSize="3rem" />
                                
                                ) :(
                                    <Avatar name={props.name} src="https://bit.ly/broken-link" />
                                )
                                }


                                    <Text color={kSecondaryBlueLight}>
                                        {name}
                                    </Text>
                    if({props.visit}===true){
                                        <Textarea placeholder="Enter reason for skill Validation" color="black.400" />
                                    }
                                    <ModalCloseButton />
                                </Stack>
                            </stack>
                            <ModalFooter>
                                <Button colorScheme="blue" mr={3} onClick={()=> handleValidation()}>
                                    Post
            </Button>
                            </ModalFooter>
                        </Box>
                    </ModalContent>

                </Modal>
                </>
)}

                  
                 </HStack>
                    </Box>
                    </Flex>
                    </>
                
                  
                    
    
    );
}


}
Skill.propTypes = {
    skillname: PropTypes.string,
    validations:PropTypes.integer,
    visit:PropTypes.bool
  }
  Skill.defaultProps =   {
      
      visit:"1",
      validations:"0"
  }
  export default Skill