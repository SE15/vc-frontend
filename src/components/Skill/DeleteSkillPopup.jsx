
import {
    Text,
    IconButton,
    useDisclosure,
    Alert, 
    AlertIcon
} from "@chakra-ui/react";

import {
    DeleteIcon
} from '@chakra-ui/icons';

import PopupWindow from '../ContainerTemplates/PopupWindow';

const DeleteSkillPopup = ({ name, onClick, isLoading }) => {
    const { isOpen, onOpen, onClose } = useDisclosure();

    return (
        <>
            <IconButton
                variant="ghost"
                colorScheme="red"
                icon={<DeleteIcon />}
                onClick={onOpen} />
            <PopupWindow
                title="Remove Skill"
                buttonName="Remove"
                onClick={onClick(onClose)}
                isLoading={isLoading}
                isOpen={isOpen}
                onClose={onClose}>
                <Text pb={2}>Do you wish to remove <Text as="em" color="purple.700">{name}</Text> from your skills?</Text>
                <Alert status="warning" borderRadius="1rem" h="25%" w="100%" pt={2}>
                    <AlertIcon />
                    You cannot undo this action once you remove the skill
                </Alert>
            </PopupWindow>
        </>
    );
}

export default DeleteSkillPopup;