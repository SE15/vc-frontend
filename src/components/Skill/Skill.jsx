
import {
    Box,
    HStack,
    Text,
    IconButton,
    Tooltip,
    Spacer,
    useDisclosure,
    Alert, 
    AlertIcon
} from "@chakra-ui/react";

import {
    DeleteIcon,
    ArrowUpIcon
} from '@chakra-ui/icons';

import PopupWindow from '../ContainerTemplates/PopupWindow';


const Skill = ({ validations, name, isOwner, onClick, isAuth, isLoading, id, alreadyValidated }) => {
    const icon = isOwner
        ? <DeletePopup
            name={name}
            onClick={onClick(id, name)}
            isLoading={isLoading} />
        : <ValidatePopup
            name={name}
            onClick={onClick(id, name)}
            isLoading={isLoading} 
            alreadyValidated={alreadyValidated}/>;
    const color = isOwner ? "red" : "green";

    return (
        <HStack
            w="100%"
            h="60px"
            borderWidth="1px"
            borderRadius="lg"
            px={4} py={2}
            bg="purple.100"
            boxShadow="lg"
            minWidth="350px"
        >
            <Text color="purple.700" fontWeight="bold" pt={1} pr={6}>{validations}</Text>
            <Text color="gray.700" fontWeight={500} align="left">{name}</Text>
            <Spacer />
            {isAuth ? <IconButton
                variant="ghost"
                colorScheme={color}
                icon={icon}
                onClick={onClick} />
                : <Box w={10} />}
        </HStack>
    );
}

const DeletePopup = ({ name, onClick, isLoading }) => {
    const { isOpen, onOpen, onClose } = useDisclosure();

    return (
        <>
            <IconButton
                variant="ghost"
                colorScheme="red"
                icon={<DeleteIcon />}
                onClick={onOpen} />
            <PopupWindow
                title="Remove Skill"
                buttonName="Remove"
                onClick={onClick(onClose)}
                isLoading={isLoading}
                isOpen={isOpen}
                onClose={onClose}>
                <Text pb={2}>Do you wish to remove <Text as="em">{name}</Text> from your skills?</Text>
                <Alert status="warning" borderRadius="1rem" h="25%" w="100%" pt={2}>
                    <AlertIcon />
                    You cannot undo this action once you remove the skill
                </Alert>
            </PopupWindow>
        </>
    );
}

const ValidatePopup = ({ name, onClick, isLoading, alreadyValidated }) => {
    const { isOpen, onOpen, onClose } = useDisclosure();

    return (
        <>
            <IconButton
                variant="ghost"
                colorScheme="green"
                icon={<Tooltip label={`${alreadyValidated ? 'Already validated' : 'Validate' }`} fontSize="md"><ArrowUpIcon /></Tooltip>}
                onClick={onOpen}
                isDisabled={alreadyValidated} />
            <PopupWindow
                title="Validate Skill"
                buttonName="Validate"
                onClick={onClick(onClose)}
                isLoading={isLoading}
                isOpen={isOpen}
                onClose={onClose}>
                <Text pb={2}>Do you wish to validate the skill, <Text as="em">{name}</Text>?</Text>
                <Alert status="warning" borderRadius="1rem" h="25%" w="100%" pt={2}>
                    <AlertIcon />
                    You cannot undo this action once you validate the skill
                </Alert>
            </PopupWindow>
        </>
    );
}

export default Skill;