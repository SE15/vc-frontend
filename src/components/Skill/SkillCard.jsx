
import {
    Box,
    HStack,
    Text,
    IconButton,
    Tooltip,
    Spacer
} from "@chakra-ui/react";

import {
    DeleteIcon,
    ArrowUpIcon
} from '@chakra-ui/icons';

const SkillCard = ({ validations, name, isOwner, onClick, isAuth }) => {
    const icon = isOwner ? <DeleteIcon />
        : <Tooltip label="Validate" fontSize="md">
            <ArrowUpIcon />
        </Tooltip>;
    const color = isOwner ? "red" : "green";

    return (
            <HStack 
            overflow="hidden" 
            w="100%" 
            h="60px" 
            borderWidth="1px" 
            borderRadius="lg" 
            px={4} py={2} 
            bg="purple.100" 
            boxShadow="lg"
            minWidth="350px"
            >
                <Text color="purple.700" fontWeight="bold" pt={1} pr={6}>{validations}</Text>
                <Text color="gray.700" fontWeight={500} align="left">{name}</Text>
                <Spacer />
                {isAuth ? <IconButton
                    variant="ghost"
                    colorScheme={color}
                    icon={icon} 
                    onClick={onClick}/>
                : <Box w={10}/>}
            </HStack>
    );
}

export default SkillCard;