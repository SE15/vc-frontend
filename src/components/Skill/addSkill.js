import { React, useState } from "react"

import PropTypes from "prop-types"
import {
    Button, Modal, ModalOverlay, ModalContent, Image
    , ModalCloseButton, Text, useDisclosure, ModalFooter,
     Textarea, Avatar, Stack, Heading, Box
} from '@chakra-ui/react';
import { kPrimaryBlack, kPrimaryBlackLight,kPrimaryGray, kSecondaryBlue, kSecondaryBlueLight } from '../../utils/constants'


const AddSkill = props => {
    return (
        <>
            <AddSkillButton onClickAddSkill={props.onClickAddSkill}/>
        </>
    )
}


function AddSkillButton(props) {
    const [isLoggedIn, setIsLoggedIn] = useState(true);
    const { isOpen, onOpen, onClose } = useDisclosure()
    const [isLoading, setIsLoading] = useState(false);
    const [error, setError] = useState('');

    const onClickAddSkill = async (event, skill)=>{
        var s = event.target.value();
        setIsLoading(true);
        try{
            onClose();
            props.onClickAddSkill(s);
            setIsLoading(false);

        }catch{
            setIsLoading(false);
            setError("Couldn't add skill")
        }
    }

    return (
        <>
            <Box>
                <Button onClick={onOpen} bg={kSecondaryBlueLight}>Add Skill</Button>
            </Box>
            <Modal isOpen={isOpen} onClose={onClose} color="black.700" closeOnOverlayClick="false">
                <ModalOverlay />
                <ModalContent>
                    <Box bg={kPrimaryBlackLight} w="100%" p={4} color="white">
                        <Stack>
                            <Heading color="black.400" fontSize="lg">
                                Add new Skill
                            </Heading>
                            <br />
                            <Stack direction="row">
                
                                <Textarea placeholder="Enter Recommendation" color="black.400" />
                                
                                <ModalCloseButton />
                            </Stack>
                        </Stack>
                        <ModalFooter>
                            <Button colorScheme="blue" mr={3} onClick={()=> onClickAddSkill()}>
                                Add
                            </Button>
                        </ModalFooter>
                    </Box>
                </ModalContent>
            </Modal>
        </>
    );

}

AddSkill.propTypes = {
    name: PropTypes.string,
    image: PropTypes.string,
    visit: PropTypes.bool
}
AddSkill.defaultProps = {
    image: "null",
    visit: true
}
export default AddSkill;  

