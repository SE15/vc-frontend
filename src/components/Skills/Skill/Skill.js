
//import { Link } from "react-router-dom"

import { Spacer, Heading, Flex, Button, Box, HStack ,CircularProgress} from "@chakra-ui/react";
import { DeleteIcon,AddIcon} from '@chakra-ui/icons';
// import {
//     Modal, ModalOverlay, ModalContent, Image
//     , ModalCloseButton, Text, ModalFooter,
//      Textarea, Avatar, Stack
// } from '@chakra-ui/react';
import {
    AlertDialog,Badge,
    AlertDialogBody,
    AlertDialogFooter,
    AlertDialogHeader,
    AlertDialogContent,
    AlertDialogOverlay 
} from "@chakra-ui/react";

import React, { Component } from 'react'

import axios from 'axios';
import { kPrimaryBlack, kPrimaryBlackLight,kPrimaryGray, kSecondaryBlue, kSecondaryBlueLight } from './../../../constants';


class Skill extends Component {
    constructor(props) {
        super(props);
        this.state = { 
            isValidated:false,
            key: props.key,
            isOpen: false,
            name: props.name,
            image: props.image,
            isLoading: false,
            validation: props.validations,
            skillname: props.skillname,
            error: ''
         };
    }

    onClose(){
        this.setState({
            ...this.state,
            isOpen: false
        }); 
    }
    onOpen(){
        this.setState({
            ...this.state,
            isOpen: true
        }); 
    }

    handleValidation= event=>{
        console.log(this.state.validation + 1)
        // this.setState({
        //     ...this.state,
        //     isLoading: true
        // });

        try{
            var validations = this.state.validation;
            //need to create correct api call
            //const res=axios.put('api/users/skills/:id',{validations})
            this.setState({
                ...this.state,
                validation: validations+1,
                isLoading: false,
                isValidated: true
            });

        }catch{
            this.setState({
                ...this.state,
                validation: this.validation,
                isLoading: false,
                error: "Couldn't validate skill"
            });
        }
    }

    deleteSkill = () => {
        if (this.state.skillname!=undefined) {
            console.log(this.state.skillname);
            this.onClose();
            this.props.onClick(this.props.skillname);
        }
    }

    cancelRef = React.createRef();

    render() {
        
        if(this.props.visit==false){             
            return (
                <>
                    <Flex>
                        <Box w="390px" p={4} bg={kPrimaryGray}>
                            <HStack  spacing="3rem">
                                    
                                <Badge ml="1" fontSize="lg" colorScheme="blue" variant="outline" borderRadius="5rem">
                                    {this.state.validation}
                                </Badge>
                                    
                                <Heading size="md" color={kSecondaryBlue}>{this.state.skillname}</Heading>
                                
                                <Spacer />
                            
                                <Button id="deleteBtn"rightIcon={<DeleteIcon />} colorScheme="red"  onClick={this.onOpen.bind(this)}></Button>
                            
                                <AlertDialog
                                    isOpen={this.state.isOpen}
                                    leastDestructiveRef={this.cancelRef}
                                    onClose={this.onClose.bind(this)}
                                >
                                    <AlertDialogOverlay>
                                    <AlertDialogContent>
                                        <Box bg={kPrimaryBlackLight} w="100%" p={4} color="white">
    
                                            <AlertDialogHeader fontSize="lg" fontWeight="bold">
                                                Delete Skill
                                            </AlertDialogHeader>
    
                                            <AlertDialogBody>
                                                Are you sure? You can't undo this action afterwards.
                                            </AlertDialogBody>
    
                                            <AlertDialogFooter>
                                            
                                            <Button colorScheme={kPrimaryBlack} ref={this.cancelRef} onClick={this.onClose.bind(this)}>
                                                Cancel
                                            </Button>
                                            <Button colorScheme="red" onClick={this.deleteSkill.bind(this)} ml={3}>
                                                Delete
                                            </Button>
                                            </AlertDialogFooter>
    
                                        </Box>
    
                                    </AlertDialogContent>
                                    </AlertDialogOverlay>
                                </AlertDialog>
                            </HStack>
                        </Box>
                    </Flex>
                </>  
            );
        }else{
            return (
                <>
                    <Flex>
                        <Box w="390px" p={4} bg={kPrimaryGray}>
                            <HStack  spacing="3rem">
                                <Badge ml="1" fontSize="0.8em" colorScheme="blue" variant="outline" borderRadius="5rem">
                                    {this.state.validation}
                                </Badge>
                                
                                <Heading size="md" color={kSecondaryBlue}>{this.state.skillname}</Heading>
                            
                                <Spacer />
                        
                                {
                                    this.state.isLoading ? (
                                        <CircularProgress isIndeterminate color="green.300"/>
                        
                                    ) :(
                                        <>
                                            <Button id="addBtn" isDisabled={this.state.isValidated? true:false}  size="sm" rightIcon={<AddIcon />} colorScheme="blue" onClick={this.handleValidation.bind(this)}>
                                            </Button>
                                        </>
                                    )
                                }
                            </HStack>
                        </Box>
                    </Flex>
                </>            
            );
        }
    }
}

export default Skill;