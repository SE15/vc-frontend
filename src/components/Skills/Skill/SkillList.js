import Skill from './Skill'
const SkillList=(props)=>{
    console.log('skill list'); 
return props.SkillList.map((skill,index)=>{
    return (
    <Skill
    name={skill.skillname}
    validations={skill.validations}
    visit={skill.visit}/>
    );
});
}
export default SkillList