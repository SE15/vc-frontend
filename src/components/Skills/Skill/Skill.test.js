import React from 'react';
import { Button,AddIcon,DeleteIcon} from '@chakra-ui/icons';

import {configure,mount,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'

import Skill from './Skill'

configure({adapter:new Adapter()});

describe('<Skill />',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper =shallow(<Skill name='Anne De Mel' validations={5} />);
    });
    it('should set isValidated to true when validated',()=>{
        
        expect(wrapper.instance().state.isValidated).toBe(false);
        

        wrapper.instance().handleValidation()
        expect(wrapper.instance().state.isValidated).toBe(true);
       
    });

    it('should increase validations when validated',()=>{
        
        expect(wrapper.instance().state.validation).toBe(5);
        

        wrapper.instance().handleValidation()
        expect(wrapper.instance().state.validation).toBe(6);
       
    });
    
    it('should render red colour button if its not a visit by other user',()=>{
        wrapper.setProps({visit:false})
        expect(wrapper.getElements()).toMatchSnapshot();
    });
    
    it('should render blue colour button if its a visit by other user',()=>{
        wrapper.setProps({visit:true})
        expect(wrapper.getElements()).toMatchSnapshot();
    });

    it('should increase validations and disable add button when validated',()=>{
        wrapper.setProps({visit:true})
        wrapper.instance().handleValidation();
        expect(wrapper.getElements()).toMatchSnapshot();
    });

    it('should keep add button disabled if already validated',()=>{
        wrapper.setProps({visit:true})
        wrapper.setState({isValidated:true})
        expect(wrapper.getElements()).toMatchSnapshot();
    })

   
});