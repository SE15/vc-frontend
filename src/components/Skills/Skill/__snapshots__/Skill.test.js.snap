// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skill /> should render blue colour button if its a visit by other user 1`] = `
Array [
  <React.Fragment>
    <Flex>
      <Box
        bg="#E2E8F0"
        p={4}
        w="390px"
      >
        <HStack
          spacing="3rem"
        >
          <Badge
            borderRadius="5rem"
            colorScheme="blue"
            fontSize="0.8em"
            ml="1"
            variant="outline"
          >
            5
          </Badge>
          <Heading
            color="#01579b"
            size="md"
          />
          <Spacer />
          <React.Fragment>
            <Button
              colorScheme="blue"
              id="addBtn"
              isDisabled={false}
              onClick={[Function]}
              rightIcon={<AddIcon />}
              size="sm"
            />
          </React.Fragment>
        </HStack>
      </Box>
    </Flex>
  </React.Fragment>,
]
`;

exports[`<Skill /> should increase validations and disable add button when validated 1`] = `
Array [
  <React.Fragment>
    <Flex>
      <Box
        bg="#E2E8F0"
        p={4}
        w="390px"
      >
        <HStack
          spacing="3rem"
        >
          <Badge
            borderRadius="5rem"
            colorScheme="blue"
            fontSize="0.8em"
            ml="1"
            variant="outline"
          >
            6
          </Badge>
          <Heading
            color="#01579b"
            size="md"
          />
          <Spacer />
          <React.Fragment>
            <Button
              colorScheme="blue"
              id="addBtn"
              isDisabled={true}
              onClick={[Function]}
              rightIcon={<AddIcon />}
              size="sm"
            />
          </React.Fragment>
        </HStack>
      </Box>
    </Flex>
  </React.Fragment>,
]
`;

exports[`<Skill /> should keep add button disabled if already validated 1`] = `
Array [
  <React.Fragment>
    <Flex>
      <Box
        bg="#E2E8F0"
        p={4}
        w="390px"
      >
        <HStack
          spacing="3rem"
        >
          <Badge
            borderRadius="5rem"
            colorScheme="blue"
            fontSize="0.8em"
            ml="1"
            variant="outline"
          >
            5
          </Badge>
          <Heading
            color="#01579b"
            size="md"
          />
          <Spacer />
          <React.Fragment>
            <Button
              colorScheme="blue"
              id="addBtn"
              isDisabled={true}
              onClick={[Function]}
              rightIcon={<AddIcon />}
              size="sm"
            />
          </React.Fragment>
        </HStack>
      </Box>
    </Flex>
  </React.Fragment>,
]
`;

exports[`<Skill /> should render red colour button if its not a visit by other user 1`] = `
Array [
  <React.Fragment>
    <Flex>
      <Box
        bg="#E2E8F0"
        p={4}
        w="390px"
      >
        <HStack
          spacing="3rem"
        >
          <Badge
            borderRadius="5rem"
            colorScheme="blue"
            fontSize="lg"
            ml="1"
            variant="outline"
          >
            5
          </Badge>
          <Heading
            color="#01579b"
            size="md"
          />
          <Spacer />
          <Button
            colorScheme="red"
            id="deleteBtn"
            onClick={[Function]}
            rightIcon={<DeleteIcon />}
          />
          <AlertDialog
            isOpen={false}
            leastDestructiveRef={
              Object {
                "current": null,
              }
            }
            onClose={[Function]}
          >
            <ModalOverlay>
              <ForwardRef>
                <Box
                  bg="#4f5b62"
                  color="white"
                  p={4}
                  w="100%"
                >
                  <ModalHeader
                    fontSize="lg"
                    fontWeight="bold"
                  >
                    Delete Skill
                  </ModalHeader>
                  <ModalBody>
                    Are you sure? You can't undo this action afterwards.
                  </ModalBody>
                  <ModalFooter>
                    <Button
                      colorScheme="#263238"
                      onClick={[Function]}
                    >
                      Cancel
                    </Button>
                    <Button
                      colorScheme="red"
                      ml={3}
                      onClick={[Function]}
                    >
                      Delete
                    </Button>
                  </ModalFooter>
                </Box>
              </ForwardRef>
            </ModalOverlay>
          </AlertDialog>
        </HStack>
      </Box>
    </Flex>
  </React.Fragment>,
]
`;
